sdk/src/main/scala/com/horizen/account/receipt/EthereumLog.scala

```diff
@@ -0,0 +1,106 @@
+package com.horizen.account.receipt
+
+import com.horizen.evm.utils.{Address, Hash}
+import com.horizen.utils.BytesUtils
+import scorex.core.serialization.{BytesSerializable, ScorexSerializer}
+import scorex.util.serialization.{Reader, Writer}
+
+import java.util
+
+case class EthereumLog(
+                         consensusDataLog : EthereumConsensusDataLog,
+                         transactionHash: Array[Byte],
+                         transactionIndex: Int,
+                         blockHash: Array[Byte],
+                         blockNumber: Int,
+                         logIndex : Int,
+                         removed : Int
+                       ) extends BytesSerializable  {
+
+
+  override type M = EthereumLog
+
+  override def serializer: ScorexSerializer[EthereumLog] = EthereumLogSerializer
+
+  def this(consensusDataLog : EthereumConsensusDataLog) {
+    this(consensusDataLog, new Array[Byte](0), -1, new Array[Byte](0), -1, -1, -1)
+  }
+
+  def update(txHash: Array[Byte], txIndex: Int, blHash: Array[Byte], blNumber: Int, idx: Int): EthereumLog =
+    EthereumLog(this.consensusDataLog, txHash, txIndex, blHash, blNumber, idx, removed=0)
+
+  override def toString: String = {
+
+    var txHashStr : String = "null"
+    var blockHashStr: String = "null"
+
+    if (transactionHash != null)
```

Comment:
can it be null? same for `blockHash`

Reply:
Should not be, but for the time being I left that check